packages/grid/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -44,8 +44,6 @@ function GridOverlayWrapper(props: React.PropsWithChildren<{}>) {
         position: 'absolute',
```

Comment:
We removed `pointerEvents: none` but this was not sufficient to fix #5725. Since we didn't add `zIndex`, the overlay stays behind the virtual scroller because it has `position: relative`, while the overlay has `position: absoulte`. We need to put the overlay on top of everything to fix #5725 too

```suggestion
        position: 'relative',
```

In https://codesandbox.io/s/mui-x-norowsoverlay-bug-forked-g6npfv?file=/src/App.tsx I still can't click the button.

Reply:
One solution that would keep https://github.com/mui/mui-x/blob/master/test/e2e/index.test.ts#L330 passing is to render the overlays inside the virtualization component.

Reply:
Maybe it will be possible to do without breaking changes - I'll take a look